Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Call.hs

```diff
@@ -50,13 +50,20 @@ directCallStatusCallback callSid dialCallStatus recordingUrl_ callDuration = do
   runTransaction $ QCallStatus.updateCallStatus callStatus.id (exotelStatusToInterfaceStatus dialCallStatus) (fromMaybe 0 callDuration) recordingUrl
   return Ack
 
-getCustomerMobileNumber :: (CacheFlow m r, EsqDBFlow m r, EsqDBReplicaFlow m r, EncFlow m r) => Text -> Text -> Text -> ExotelCallStatus -> m GetCustomerMobileNumberResp
-getCustomerMobileNumber callSid callFrom_ callTo_ callStatus = do
+getCustomerMobileNumber :: (CacheFlow m r, EsqDBFlow m r, EsqDBReplicaFlow m r, EncFlow m r) => Text -> Text -> Text -> Text -> ExotelCallStatus -> m GetCustomerMobileNumberResp
+getCustomerMobileNumber callSid callFrom_ callTo_ dtmfNumber callStatus = do
   let callFrom = dropFirstZero callFrom_
   let callTo = dropFirstZero callTo_
   mobileNumberHash <- getDbHash callFrom
   exophone <- CQExophone.findByPhone callTo >>= fromMaybeM (ExophoneDoesNotExist callTo)
-  driver <- runInReplica $ QPerson.findByMobileNumberAndMerchant "+91" mobileNumberHash exophone.merchantId >>= fromMaybeM (PersonWithPhoneNotFound callFrom)
+  checkMobNo <- runInReplica $ QPerson.findByMobileNumberAndMerchant "+91" mobileNumberHash exophone.merchantId
+  driver <- case checkMobNo of
+    Nothing -> do
+      dtmfMobileHash <- getDbHash dtmfNumber
```

Comment:
use dropFirstZero function on this number first

Reply:
Also you can calculate dtmf's hash first and then use bind operator in line 59 itself instead of creating case stmt. You can return dtmfNumber too. Something like below should work

```
(checkMobNo, usedDtmfNumber) <- runInReplica $ QPerson.findByMobileNumberAndMerchant "+91" mobileNumberHash 
exophone.merchantId >>= \case
      Nothing -> do
           --calculations--
           return (person, dtmfNumber)
      Just entity -> (entity, Nothing)
```

Reply:
Also you can use bind operator in line 59 itself instead of creating case stmt. You can return dtmfNumber too. Something like below should work

```
(checkMobNo, usedDtmfNumber) <- runInReplica $ QPerson.findByMobileNumberAndMerchant "+91" mobileNumberHash 
exophone.merchantId >>= \case
      Nothing -> do
           --calculations--
           return (person, dtmfNumber)
      Just entity -> return (entity, Nothing)
```

Reply:
It is not confirm currently where this number will come along with leading zero or not. It needs to be checked in dashboard.

Reply:
You may still use it

Reply:
Sure